pipeline/data_merging/brca_pseudonym_generator.py

```diff
@@ -165,89 +168,153 @@ def get_synonyms(x, hgvs_proc, syn_ac_dict):
+    """ Merging with synonmys already determined from sources and cleaning up"""
+    orig_list = [s for s in row[SYNONYMS_COL].split(',') if s]  # filter away ''
 
-    combined = set(orig_list + x[NEW_SYNONYMS_FIELD])
+    combined = set(orig_list + row[NEW_SYNONYMS_FIELD])
     list_sorted = sorted(list(combined))
-    return ','.join(list_sorted)
+
+    # remove redundant representations, which we are already included in other columns
+    reps_other_cols = {row[GENOMIC_HGVS_HG37_COL], row[GENOMIC_HGVS_HG38_COL], row[PYHGVS_CDNA_COL], row[PYHGVS_PROTEIN_COL],
+                       row[HGVS_CDNA_COL]}
+    list_sorted_cleaned = [v for v in list_sorted if v not in reps_other_cols and v != str(None) and v != '-']
+
+    return ','.join(list_sorted_cleaned)
 
 
 @click.command()
 @click.argument('input', type=click.Path(readable=True))
 @click.argument('output', type=click.Path(writable=True))
 @click.option('--log-path', default='pseudonym_generator.log', help="Log file pth")
-@click.option("--pkl", help="Saving HGVS cDNA objects to save time during development")
 @click.option("--config-file", required=True, help="path to gene configuration file")
 @click.option('--resources', help="path to directory containing reference sequences")
-def main(input, output, pkl, log_path, config_file, resources):
+@click.option('--processes', type=int, help='Number of processes to use for parallelization', default=8)
+def main(input, output, log_path, config_file, resources, processes):
     utils.setup_logfile(log_path)
 
     cfg_df = config.load_config(config_file)
 
-    syn_ac_dict = { x[config.SYMBOL_COL] : x[config.SYNONYM_AC_COL].split(';') for _, x in cfg_df.iterrows()}
-    cdna_default_ac_dict = { x[config.SYMBOL_COL] : x[config.HGVS_CDNA_DEFAULT_AC] for _, x in cfg_df.iterrows()}
+    syn_ac_dict = {r[config.SYMBOL_COL]: r[config.SYNONYM_AC_COL].split(';') for _, r in cfg_df.iterrows()}
+    cdna_default_ac_dict = {r[config.SYMBOL_COL]: r[config.HGVS_CDNA_DEFAULT_AC] for _, r in cfg_df.iterrows()}
+    strand_dict = { r[config.SYMBOL_COL] : r[config.STRAND_COL] for _, r in cfg_df.iterrows() }
 
     hgvs_proc = HgvsWrapper()
 
+    logging.info("Loading data from {}".format(input))
     df = pd.read_csv(input, sep='\t')
 
-    df[VAR_OBJ_FIELD] = df.apply(lambda x: VCFVariant(x[CHR_COL], x[POS_COL], x[REF_COL], x[ALT_COL]), axis=1)
+    df[VAR_OBJ_FIELD] = df.apply(lambda r: VCFVariant(r[CHR_COL], r[POS_COL], r[REF_COL], r[ALT_COL]), axis=1)
+
+    logging.info("Converting variants to hgvs objects")
+
+    df[TMP_HGVS_HG38] = df[VAR_OBJ_FIELD].apply(
+        lambda v: v.to_hgvs_obj(hgvs_proc.contig_maps[HgvsWrapper.GRCh38_Assem]))
+
+    logging.info("Normalize genomic representation")
+
+    df = utils.parallelize_dataframe(df, _normalize_genomic_fnc(TMP_HGVS_HG38, TMP_HGVS_HG38, True, strand_dict), processes)
+    df = utils.parallelize_dataframe(df, _normalize_genomic_fnc(TMP_HGVS_HG38, TMP_HGVS_HG38_LEFT_ALIGNED, False, strand_dict),
```

Comment:
i found this a bit confusing, could use a comment to explain why normalize_genomic_fnc needs to be run twice like this

Reply:
Once we align right wrt to cDNA and once we align left (for the synonyms)